Add jitter and loss measurements to iperf3 integration

jasonrogena opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->
No breaking changes

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

To allow to better test network connectivity, add jitter and loss to the list of measurements collected by the iperf3 integration. These two additional measurements are only collected if the protocol selected is udp.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

jasonrogena committed:
Add jitter and loss measurements to iperf3 integration

To allow to better test network connectivity, add jitter and loss to the list of measurements collected by the iperf3 integration.

emontnemery commented on homeassistant/components/iperf3/__init__.py:
```diff
@@ -66,6 +70,22 @@
+    ),
+    SensorEntityDescription(
+        key=ATTR_LOSS,
+        name=ATTR_LOSS.capitalize(),
```
I think "Packet loss" would be a better name than "Loss"

emontnemery replied:
```suggestion
        name="Packet loss",
```

emontnemery commented on homeassistant/components/iperf3/__init__.py:
```diff
@@ -66,6 +70,22 @@
+        name=ATTR_LOSS.capitalize(),
+        icon="mdi:lan-disconnect",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=PERCENTAGE,
```
```suggestion
        device_class=None,
```

jasonrogena committed:
Delete the iper3 client after result gotten

To reduce the number of open files as a result of iper3 tests, del
the client after the iperf3 results have been gotten.

jasonrogena commented:
@emontnemery good for a rereview.

frenck commented on homeassistant/components/iperf3/__init__.py:
```diff
@@ -66,6 +72,22 @@
+        name="Packet Loss",
+        icon="mdi:lan-disconnect",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=None,
```
That is the default

```suggestion
```

frenck commented on homeassistant/components/iperf3/__init__.py:
```diff
@@ -66,6 +72,22 @@
+    SensorEntityDescription(
+        key=ATTR_JITTER,
+        name=ATTR_JITTER.capitalize(),
+        icon="mdi:timer",
```
This has a device class, which provides the default icon

So! The icon can be removed
